app/mailers/generic_mailer.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class GenericMailer < ApplicationMailer
+  default from: %("Mailgun Sandbox" \
+    <postmaster@sandbox8f696e611a6e4906a977c007f8971322.mailgun.org>)
+
+  def drop_duty(duty)
+    @duty = duty
+    mail(to: all_users_with_name,
+         subject: 'Duty notification: '\
+                  "#{duty.time_range.start_time.strftime('%H%M')}-"\
+                  "#{duty.time_range.end_time.strftime('%H%M')} on"\
+                  "#{duty.date.strftime('%a, %d %b %Y')} at"\
+                  "#{duty.place.name}")
+  end
+
+  def problem_report(problem)
+    @problem = problem
+    mail(to: cell_users_with_name('Technical'),
```

Comment:
Do something like `User.has_role('tech')` or something like that

Reply:
Also, ideally instead of abstracting out the `users_with_name`, what you want is to abstract the mailing function instead, so you can just pass in the list of User objects you wish to email.

Reply:
Hmm... not sure what you mean by abstracting the mailing function? Cos if I do it would just be a bit silly
```ruby
def mail_abstraction(users, subject)
  mail(to: users&.pluck(:username, :email)&.map { |u|  %("#{u[0]}" <#{u[1]}>) },
       subject: subject)
end
```